ReactWindows/ReactNative/Views/TextInput/ReactPasswordBoxShadowNode.cs

```diff
@@ -0,0 +1,334 @@
+﻿using Facebook.CSSLayout;
+using ReactNative.Bridge;
+using ReactNative.Reflection;
+using ReactNative.UIManager;
+using ReactNative.UIManager.Annotations;
+using ReactNative.Views.Text;
+using System;
+using Windows.Foundation;
+using Windows.UI.Text;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Documents;
+using Windows.UI.Xaml.Media;
+
+namespace ReactNative.Views.TextInput
+{
+    /// <summary>
+    /// This extension of <see cref="LayoutShadowNode"/> is responsible for 
+    /// measuring the layout for Native <see cref="PasswordBox"/>.
+    /// </summary>
+    public class ReactPasswordBoxShadowNode : LayoutShadowNode
+    {
+        private const int Unset = -1;
+
+        private float[] _computedPadding;
+
+        private int _letterSpacing;
+        private int _numberOfLines;
+
+        private double _fontSize = Unset;
+        private double _lineHeight;
+
+        private FontStyle? _fontStyle;
+        private FontWeight? _fontWeight;
+        private TextAlignment _textAlignment = TextAlignment.DetectFromContent;
+
+        private string _fontFamily;
+        private string _password;
+
+        private int _jsEventCount = Unset;
+
+        /// <summary>
+        /// Instantiates the <see cref="ReactPasswordBoxShadowNode"/>.
+        /// </summary>
+        public ReactPasswordBoxShadowNode()
+        {
+            var computedPadding = GetDefaultPaddings();
+            SetPadding(CSSSpacingType.Left, computedPadding[0]);
+            SetPadding(CSSSpacingType.Top, computedPadding[1]);
+            SetPadding(CSSSpacingType.Right, computedPadding[2]);
+            SetPadding(CSSSpacingType.Bottom, computedPadding[3]);
+            MeasureFunction = MeasureTextInput;
+        }
+
+        /// <summary>
+        /// Sets the password for the node.
+        /// </summary>
+        /// <param name="password">The password.</param>
+        [ReactProp("password")]
```

Comment:
does this have to be called "password" or can we just call it "text" for consistency with TextInput?